pkg/ocm/clustertransfer/cluster_transfer.go

```diff
@@ -0,0 +1,136 @@
+package clustertransfer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/insights/insightsclient"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type Controller struct {
+	coreClient   corev1client.CoreV1Interface
+	ctx          context.Context
+	configurator Configurator
+	client       *insightsclient.Client
+}
+
+// Configurator represents the interface to retrieve the configuration
+type Configurator interface {
+	Config() *config.Controller
+	ConfigChanged() (<-chan struct{}, func())
+}
+
+// New creates new instance of controllers
+func New(ctx context.Context, coreClient corev1client.CoreV1Interface, configurator Configurator,
+	insightsClient *insightsclient.Client) *Controller {
+	return &Controller{
+		coreClient:   coreClient,
+		ctx:          ctx,
+		configurator: configurator,
+		client:       insightsClient,
+	}
+}
+
+// Run periodically queries the OCM API and update pull-secret accordingly
+func (c *Controller) Run() {
+	cfg := c.configurator.Config()
+	endpoint := cfg.OCMConfig.ClusterTransferEndpoint
+	interval := cfg.OCMConfig.ClusterTransferInterval
+	configCh, cancel := c.configurator.ConfigChanged()
+	defer cancel()
+	for {
+		select {
+		case <-time.After(interval):
+			c.requestDataAndUpdateSecret(endpoint)
+		case <-configCh:
+			cfg := c.configurator.Config()
+			interval = cfg.OCMConfig.ClusterTransferInterval
+			endpoint = cfg.OCMConfig.ClusterTransferEndpoint
+		}
+	}
+}
+
+func (c *Controller) requestDataAndUpdateSecret(endpoint string) {
+	// TODO us insights client with authentication
+	// do the request
+	cli := &http.Client{}
+	res, err := doRequest(endpoint, cli)
+	if err != nil {
+		klog.Errorf("Failed to do request: %v", err)
+		return
+	}
+	newData, err := readResponse(res)
+	if err != nil {
+		klog.Errorf("Failed to read response data: %v", err)
+		return
+	}
+
+	err = c.updatePullSecret(newData)
+	if err != nil {
+		klog.Errorf("Failed to update pull-secret: %v", err)
+		return
+	}
+	klog.Info("pull-secret successfully updated")
+}
+
+func (c *Controller) updatePullSecret(newData []byte) error {
+	pullSecret, err := c.getPullSecret()
+	if err != nil {
+		return err
+	}
+
+	updatedData, err := jsonpatch.MergePatch(pullSecret.Data[v1.DockerConfigJsonKey], newData)
+	if err != nil {
+		return err
+	}
+
+	pullSecret.Data[v1.DockerConfigJsonKey] = updatedData
+	_, err = c.coreClient.Secrets("openshift-config").Update(c.ctx, pullSecret, metav1.UpdateOptions{})
```

Comment:
Hmm we need to be a bit smarter here - we should check if there's really some diff and update only in this case.